programs/bpf_loader/src/syscalls.rs

```diff
@@ -2976,12 +2982,26 @@ where
                 invoke_context,
             )?;
             {
-                let mut account = account.borrow_mut();
-                account.copy_into_owner_from_slice(caller_account.owner.as_ref());
-                account.set_data_from_slice(caller_account.data);
-                account.set_lamports(*caller_account.lamports);
-                account.set_executable(caller_account.executable);
-                account.set_rent_epoch(caller_account.rent_epoch);
+                callee_account
+                    .set_lamports(*caller_account.lamports)
+                    .map_err(SyscallError::InstructionError)?;
+                callee_account
+                    .set_data(caller_account.data)
+                    .map_err(SyscallError::InstructionError)?;
+                callee_account
+                    .set_executable(caller_account.executable)
+                    .map_err(SyscallError::InstructionError)?;
+                callee_account
+                    .set_owner(caller_account.owner.as_ref())
+                    .map_err(SyscallError::InstructionError)?;
+                drop(callee_account);
+                let callee_account = invoke_context
```

Comment:
This is weird, maybe callee_account should have a method to set rent_epoch that does nothing.?

Reply:
I agree and I think we need to do something about `rent_epoch` in general, but in a different PR.
We can't just do nothing / ignore `rent_epoch` here, because that would change the behavior of CPI programs (as the caller can read what the callee wrote).